### ISSUE TITLE
all my accounts and service princibles seem to have this issue- I will look up the owner or just try to list the resources even and get a failure on every id I get that associates with any of the 3 accounts I own- is there a command to force a master list or the dependency tree for all ids? maybe something is associated incorrectly

### ISSUE THREAD

**Issue by Kstuart1992** [Azure/azure-cli#22840]
## Describe the bug

**Command Name**
`az ad sp show`

**Errors:**
```
Resource '4765445b-XXXXX-.......' does not exist or one of its queried reference-property objects are not present.

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az ad sp show --id {}`

## Environment Summary
```
Linux-5.17.13-300.fc36.x86_64-x86_64-with-glibc2.35, Fedora Linux 36 (Workstation Edition)
Python 3.10.5
Installer: PIP

azure-cli 2.37.0

Extensions:
interactive 0.4.5

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```

**Comment by Kstuart1992** [comment]
Thanks by the way - if there is a way to get a top down view I can debug and write up the root cause if it wasn't simply human error. 
It works fine with the docker image that is used for my mobile access but on Win or Linux - the CLI is hitting this issue 
I was thinking one of a few things based on the verbose error returned from MS Graph: 

A token or set of GPG keys that are being checked in an improper order or just need to be renewed [I am looking to this now]

If there is a core account 'sys' file, ie USING dbAzureAccount_A SELECT * FROM sys.index_AcctIDs - in that KQL arrangement Azure uses please advise - ideally if it brings back the ids associated to the account is some hierarchical or tree based result or table [unsure of admin / sys commands that work in az cli] 

An exemption is not present to allow the raised access of the former query being resolved to bypass a standard privilege or to know not to repeat the query sense it has already resolved - in the case of a redirect between accounts - for this I'm guessing the source code to check for this logic would be in MS Graph rather than Azure - unless it is a separate protocol in an Azure module [ if you code advise me on that I can begin there ]

The issue from the --debug logs looks consistent: send GET request once query resolves the Auth and checks privileges to access resource in question. This resolves but then preforms a redirect [protocol unknown] and issues a reference id to the original query, this redirect fails each time, where the error above is then returned without any other detail, the python script only recognizes the 404 message the now redirected [unknown, perhaps authoritative account or other - some type of loop is generated between accounts - this is the root cause in any case] directory is queried for the originally resolved one and returns the 404 because its simply not there, without an exemption to check the prior result- it does its python thing, returning a print message and logging the steps from - to for debug and exits 

So if I am far off in my understanding of the possible issue an d you can direct me to the appropriate docs/code to begin, I'll be very thankful - I can send any fix in relation if one exists or if its strictly human error - I can close the issue here- My thanks

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by jiasli** [comment]
I can't really understand what you are talking about.

If you do see any problem with `az ad sp show`, you may run the command with `--debug` flag, it will print the raw HTTP requests for Graph service and help you troubleshoot your usage or service issue. You may also share the output in this issue as a comment.

**Comment by Kstuart1992** [comment]
You got it - don't let this sit on your issue board. Close this one and if it's something appropriate I'll send it over. I tracked down the resources I was asking about and will circle back if needed or it was interesting enough to offer the root cause. I appreciate your time getting back to me in any case. It's nothing to run step recorder over - I learned a long time ago, have a back-up, it costs less, and yaml is good anywhere =P * cheers
